### ISSUE TITLE
Page refresh due to music on old-ish iOS devices

### ISSUE THREAD

**Issue by Telecs** [Scirra/Construct-bugs#6671]
## Problem description

On iOS devices, in Safari, if I play several songs one after the other, the page refreshes by itself after a certain number of tracks played. 
I have the impression that the webm decoder does not unload the music once played.

## Attach a .c3p

[https://pestorosso.com/music-test.c3p](https://pestorosso.com/music-test.c3p)

## Steps to reproduce

1. Launch with an old-ish iOS device (iPad Gen6 or older)
2. Tap the screen
3. Wait around 1 minute to see the page refresh itself (sorry, I didn't manage to shorten it further). The time it takes depends of the device you use.

## Observed result

The page refresh itself.
Doesn't happen on any android / Windows I checked.

## Expected result

The game should just keep going on. As it is the case with low-end devices that are not on iOS. I also tested with a Wiko with 1 Gb of RAM, and I don't have any problem on this device with the attached test.

## More details

**Affected browsers/platforms:**

iOS only, Firefox, Safari, Chrome (since they're a reskin of Safari).
Mostly viewed on iPad Gen6. Happened also on a Macbook Pro 2011

On an iPad Pro, the page didn't refresh, but at some point, no more music is played.

**First affected release:**
It's been a few months since I struggle with this problem. I don't think that's quite recent.

## System details

It seems there's a audio crash report on MacOS. I guess it might be linked.

<details><summary>View details</summary>

[Error] [Construct] Audio: error starting playback: 
TypeError: null is not an object (evaluating 'this._audioDomHandler.GetAudioContext')
GetAudioContext — audioInstance.js:3:127
C3AudioInstance — audioInstance.js:1:243
C3WebAudioInstance — webAudioInstance.js:1:139
CreateInstance — audioBuffer.js:2:526
(fonction anonyme) — domSide.js:18:236
asyncFunctionResume
[native code]
promiseReactionJobWithoutPromise
promiseReactionJob
  (fonction anonyme) (domSide.js:23:325)
  asyncFunctionResume
  (fonction anonyme)
  promiseReactionJobWithoutPromise
  promiseReactionJob

</details>

**Comment by AshleyScirra** [comment]
What does 'old-ish iOS device' mean specifically?

If the device is running on an old version of iOS/Safari and modern versions are not affected, I would assume this is a Safari bug that was subsequently fixed.

**Comment by Telecs** [comment]
No, it doesn't depend on the device updates.
I tested this example on an iPad Gen 6 that has the latest update (16.3.1) installed. I think the problem is just more pronounced on devices with more limited RAM.
To help you see the bug appear faster, I suggested using an older device, but still supported by Apple. I actually got a device identical to the ones used by the schools using my game, which are iPad Gen 5-6.

**Comment by Telecs** [comment]
Do you still need more information?
At the moment I have to offer the game without music to my users. Would using another music format for this game be a solution until this problem is solved? Would it be for example possible to launch .mp3 tracks? I have the impression that Construct doesn't let me import anything else than WebM as it is.

**Comment by AshleyScirra** [comment]
I'm afraid this kind of issue is time-consuming to investigate, and due to the extraordinary number of issues filed lately we are very pushed for time. Please be patient.

**Comment by Telecs** [comment]
I know it's long, I'm confused about not being able to shorten the test time more, I've already tried to shorten the time needed to reproduce the bug as much as possible (now about 45s on iPad 6Gen). The problem is to put enough music in memory in a very short time for the page to refresh. I don't see how to speed up this process further since on iOS only one track can be launched at a time. If you have a direction you see for me to further increase the load on the device's memory, I'll gladly do another test in that direction.

As for an iOS bug, I'm not sure that's possible since they don't support the .webm format anyway which is the only one involved in this test. From what I gathered, it's probably the decoder you have set up to play webm on iOS that doesn't properly unload the music and sounds played.
